### PR TITLE
Fix the file path of tar.c

### PR DESCRIPTION
**Pull request by 1956425473q**
Fixed the issue of moving filename, causing strncpy to copy not the full path, Also added cmake to make it easier to compile.

### CODE REVIEW

**Comment by calccrypto** [comment]
Can you please show an example of what the change fixes?

**Comment by calccrypto** [comment]
Please give both libraries proper names within CMakeLists.txt and rename the actual output files with `set_target_properties`.

**Review by calccrypto — CHANGES_REQUESTED** [review]
Change tabs to 4 spaces

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	${PROJECT_SOURCE_DIR}
+)
+
+file(GLOB LIB
+"${PROJECT_SOURCE_DIR}/tar.c"
+)
+
+if(compile_static_libraries)
+	file(GLOB SRCS
+	"${PROJECT_SOURCE_DIR}/main.c"
```
Comment (calccrypto) [inline CMakeLists.txt:16]: Indent line continuations

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	)
+	add_library(tar STATIC ${LIB})
+	link_directories(${PROJECT_SOURCE_DIR})
+	add_executable(exec ${SRCS})
+	target_link_libraries(exec tar)
+else()
+	file(GLOB SRCS
+	"${PROJECT_SOURCE_DIR}/*"
```
Comment (calccrypto) [inline CMakeLists.txt:24]: Indent line continuations

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.6.0)
+project(tar)
+
+set(compile_static_libraries 1)
+set(compile_shared_libraries 1)
```
Comment (calccrypto) [inline CMakeLists.txt:5]: Should these be `option`s?

**Comment by 1956425473q** [comment]
> 您能否举例说明此更改修复了什么？

strncpy(entry -> name, filename + move, 100);
Here because the filename parameter is moved forward by 1, strncpy is not copying the full data of filename, for example, I want to copy the "/my_dir/my_file" file, because the filename is moved forward, causing strncpy to copy "my_dir/my_file", which is incomplete, if the name of the file does not contain the path, strncmp also only copies "y_file", instead of " my_file".

**Comment by 1956425473q** [comment]
> 您能否举例说明此更改修复了什么？

Sorry, I made a mistake.

**Pull request closed without merging**